File: app/src/main/java/com/stone/framework/bean/api/PageBean.java
package com.stone.framework.bean.api;

import java.util.List;

public class PageBean<T> {
    private List<T> records;
    private int current;
    private int pages;
    private int size;
    private int total;
    private String extra;

    public String getExtra() {
        return extra;
    }

    public void setExtra(String extra) {
        this.extra = extra;
    }

    public List<T> getRecords() {
        return records;
    }

    public void setRecords(List<T> records) {
        this.records = records;
    }

    public int getCurrent() {
        return current;
    }

    public void setCurrent(int current) {
        this.current = current;
    }

    public int getPages() {
        return pages;
    }

    public void setPages(int pages) {
        this.pages = pages;
    }

    public int getSize() {
        return size;
    }

    public void setSize(int size) {
        this.size = size;
    }

    public int getTotal() {
        return total;
    }

    public void setTotal(int total) {
        this.total = total;
    }
}

File: module_pet/src/main/java/com/stone/module/pet/bean/api/PageBean.kt
package com.stone.module.pet.bean.api

import java.io.Serializable

/**
 * desc   : 分页
 * author : stone
 * email  : aa86799@163.com
 * time   : 2019-07-12 11:27
 */

class PageBean<T> : Serializable {

    var pages: Int = 0 //共多少页
    var current: Int = 0 //当前页数
    var size: Int = 0 //当前页的size
    var total: Int = 0 //总共的 size
    var records: List<T>? = null //记录
    var extra: String? = null

}
